os/arch/arm/src/imxrt/imxrt_norflash.c

```diff
@@ -732,6 +732,21 @@ status_t imxrt_flexspi_nor_hyperbus_read(FLEXSPI_Type *base, uint32_t addr, uint
 }
 #endif
 
+
+/************************************************************************************
+ * Name: up_dumptoflash
+ *
+ * Description:
+ *   Wrapper function for flexispi write call
+ *
+ ************************************************************************************/
+#if defined(CONFIG_BOARD_CRASHDUMP)
+uint32_t up_dumptoflash(uint32_t addr, uint32_t *buffer, uint32_t bytes)
```

Comment:
if you use get_mtd_partition, then you can get mtd_dev_s, then you can use it.
struct mtd_dev_s *mtd = get_mtd_partition();
mtd->write(mtd, addr, bytes, buffer); 
You don't have to implement this api for all board...